components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteReporterService.java

```diff
@@ -41,16 +41,18 @@
 
     private final GraphiteReporter reporter;
     private final long reportingIntervalMillis;
-    private final InetSocketAddress address;
+    private final String host;
+    private final int port;
 
     private GraphiteReporterService(Builder builder) {
-        this.address = builder.inetSocketAddress;
-        MetricRegistry registry = checkNotNull(builder.registry);
+        this.host = requireNonNull(builder.host);
+        this.port = builder.port;
+        MetricRegistry registry = requireNonNull(builder.registry);
         GraphiteSender graphiteSender = Optional
                 .ofNullable(builder.graphiteSender)
-                .orElseGet(() -> new NonSanitizingGraphite(address));
+                .orElseGet(() -> new NonSanitizingGraphite(host, port));
 
-        String prefix = checkNotNull(builder.prefix);
+        String prefix = requireNonNull(builder.prefix);
```

Comment:
Are we following any pattern to decide when to use requireNonNull instead of checkNotNull?

Reply:
IIRC, `requireNonNull` was added in Java 8. The Styx project was began development in Java 7, so we were using Guava's `checkNotNull`. Since upgrading there is no longer a reason to use this third-party dependency when the functionality is the same.

Reply:
Prior to Java 8 we were using `checkNotNull` from Guava. But since then we should use `requireNonNull` in its place.